pkg/vsphere/compute/rp.go

```diff
@@ -88,3 +88,51 @@ func (rp *ResourcePool) GetCluster(ctx context.Context) (*object.ComputeResource
 
 	return object.NewComputeResource(rp.Client.Client, mrp.Owner), nil
 }
+
+func (rp *ResourcePool) GetDatacenter(ctx context.Context) (*object.Datacenter, error) {
+	dcRef, err := rp.getLowestAncestor(ctx, "Datacenter")
+
+	if err != nil || dcRef == nil {
+		log.Errorf("Unable to get datacenter ancestor of rp %s: %s", rp.Name(), err)
+		return nil, err
+	}
+
+	return object.NewDatacenter(rp.Client.Client, *dcRef), nil
+}
+
+func (rp *ResourcePool) getAncestors(ctx context.Context, inType string) ([]types.ManagedObjectReference, error) {
+	client := rp.Session.Vim25()
+
+	ancestors, err := mo.Ancestors(ctx, client, client.ServiceContent.PropertyCollector, rp.Reference())
+
+	if err != nil {
+		log.Errorf("Unable to get ancestors of rp %s: %s", rp.Name(), err)
+		return nil, err
+	}
+
+	outAncestors := make([]types.ManagedObjectReference, 0)
+	for _, ancestor := range ancestors {
+		if ancestor.Self.Type == inType {
+			a := ancestor.Self
```

Comment:
A few notes about this `for` loop:

* `a` is an ambiguous name - doesn't matter a lot currently since it's only used right below, but it will be if used elsewhere in the future.
* `ancestor.Self` is looked up twice (lines 115 and 116). I think it'd be better to declare a var for `ancestor.Self` at the top of the `for` loop, and use that inside.